modules/apps/commerce/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/util/ProductHelperImpl.java

```diff
@@ -72,10 +72,16 @@ public PriceModel getMinPrice(
 		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
 			"content.Language", locale, getClass());
 
-		return new PriceModel(
+		PriceModel priceModel = new PriceModel(
 			_language.format(
 				resourceBundle, "from-x",
 				cpDefinitionMinimumPriceCommerceMoney.format(locale), false));
+
+		if (cpDefinitionMinimumPriceCommerceMoney.isEmpty()) {
```

Comment:
looking at the code looks like you may have empty prices and not have the price on application set to true, I think we should better manage this scenario and not set price on application because no price was found

Reply:
As discussed we changed the CommerceMoney interface to manage priceOnApplication attribute and removed the check on isEmpty